lua/lean/_util.lua

```diff
@@ -218,6 +218,12 @@ function M.position_params_valid(params)
   return true
 end
 
+function M.make_position_params()
+  -- UTF-32 is used for general UI features and Lean 3 LSP; UTF16 is used for Lean 4 LSP
```

Comment:
What kind of general UI features use this? Are they not limited to Lean 3?

I'd love to limit all of this encoding manipulation to just Lean 3 -- and to just having a separate `make_position_params` for Lean 3, rather than carrying around both encodings and having lots of places choose one or the other, that seems easy to make a mistake on.